index.js

```diff
@@ -1,41 +1,29 @@
-const init = (token, { fastUrl, forceOverride } = {}) => {
-  if (forceOverride) {
-    chmln = undefined;
-  } else if (typeof chmln !== 'undefined') {
-    console.log(`The global variable \`chmln\` is already defined.\nPlease make sure you're not adding Chameleon to the page other than with this library. Alternatively, you can use the forceOverride option to remove any previous declaration.\n\ne.g.   chmln.init('token', { forceOverride: true });`);
-    return;
-  }
-
-  var d = document;
-  var w = window;
-  var t = token,
-  c="chmln",m="identify alias track clear set show on off custom help _data".split(" "),
-  i=d.createElement("script");if(w[c]||(w[c]={}),!w[c].root){w[c].accountToken=t,w[c].location=w.location.href.toString(),w[c].fastUrl=fastUrl||'https://fast.trychameleon.com/',
-      w[c].now=new Date;for(var s=0;s<m.length;s++){!function(){var t=w[c][m[s]+"_a"]=[];w[c][m[s]]=function(){t.push(arguments);};}();}
-      i.src=w[c].fastUrl+"messo/"+t+"/messo.min.js",
-      i.async=!0,d.head.appendChild(i);}
-}
 
-const identify = (id, options) => {
-  if (!chmln || !chmln.identify) {
-    console.log(`Failed to identify the user [${id}] because the chameleon.io script was not initialized yet.`);
-    return;
-  }
-
-  chmln.identify(id, options || {});
-}
+let chmln_object = window.chmln;
 
-const track = (eventName) => {
-  if (!chmln || !chmln.track) {
-    console.log(`Failed to track event [${eventName}] because the chameleon.io script was not initialized yet.`);
-    return;
+if (typeof chmln_object === 'undefined' || !chmln_object.identify) {
+  chmln_object = chmln_object || (window.chmln = {});
+  const defaultFunctions = "identify alias track clear set show on off custom help _data".split(" ");
+  for (var i = 0; i < defaultFunctions.length; i++) {
```

Comment:
```suggestion
  defaultFunctions.forEach((_, i) => {
```
suggestion: `for` -> `forEach` better readability (current for offers better performance, but since we're just iterating over ~9 items it doesn't matter much)

Reply:
+1. i would typically prefer `for` for performance reasons but when we already know the length & it's not very long, i think `forEach` is safe to use.

Reply:
I'm going to go with Kristin, and keep the `for` because of the performance. Any speed we can add to customer initialization is a win. Also, this code is small and not necessary to increase readability.